### PR TITLE
Add 'rho fact list' command.

### LINKED ISSUE #80: rho scan --show-fields capability no longer exists
The previous version of `rho` provided an output to users with the collection of possible facts that can be in the report.  This capability was "lost" in the latest version of the code base, however, the code still contains the documentation that just needs to be accessed (e.g. https://github.com/quipucords/rho/blob/master/module_utils/date_command.py#L62)

### PR DESCRIPTION
**Pull request by noahl**
'rho list-facts' now shows documentation about all of the facts that
Rho can collect.

The documentation is kept in `rho/fact_docs.py`. This change also adds
a script at `docs/generate_python_docs.py` which generates
`fact_docs.py` from the reStructured Text documentation at
`doc/command_syntax_usage.rst`, to make it easy to keep them in sync.

Closes #80.

### CODE REVIEW

File: doc/command_syntax_usage.rst [hunk 1]
```diff
@@ -14,6 +14,7 @@ There are eight rho commands:
  * ``edit`` - to modify an existing entry
  * ``clear`` - to remove any or all entries
  * ``show`` and ``list`` - to display one or more entries
+ * ``list-facts`` - to list the facts Rho can collect
```
Comment (chambridge) [inline doc/command_syntax_usage.rst]: I would prefer the command be named `rho fact list` which would fit the `rho <noun> <verb>` flow found in other commands.

**Review by chambridge — COMMENTED** [review]
Build appears to be failing with an import error. Looks like you may need to add a new dependency.

File: doc/command_syntax_usage.rst (hunk 1 above)
Comment (noahl) [reply doc/command_syntax_usage.rst]: Renamed.

**Comment by noahl** [comment]
@chambridge I fixed the requirement issue and some Makefile problems; I believe this is ready for review.

File: doc/command_syntax_usage.rst (hunk 1 above)
Comment (chambridge) [reply doc/command_syntax_usage.rst]: shouldn't the above documentation be updated?

**Comment by noahl** [comment]
Thanks! Fixed that.

File: doc/command_syntax_usage.rst [hunk 2]
```diff
@@ -14,7 +14,7 @@ There are eight rho commands:
  * ``edit`` - to modify an existing entry
  * ``clear`` - to remove any or all entries
  * ``show`` and ``list`` - to display one or more entries
- * ``list-facts`` - to list the facts Rho can collect
+ * ``fact`` - to show information about the facts Rho can collect
```
Comment (chambridge) [inline doc/command_syntax_usage.rst]: Sorry for the nit pick, but we tend to refer to `rho` in all lowercase in the docs. Just want to be consistent.

**Review by chambridge — APPROVED** [review]
Minor comment, otherwise looks good to me.

**Comment by noahl** [comment]
Thanks for taking the time to make the docs consistent.

File: doc/command_syntax_usage.rst (hunk 2 above)
Comment (noahl) [reply doc/command_syntax_usage.rst]: Fixed, thank you!

**Pull request merged**